polaris-react/src/components/Navigation/components/Item/Item.tsx

```diff
@@ -300,39 +255,15 @@ export function Item({
       ({url: firstUrl}, {url: secondUrl}) => secondUrl.length - firstUrl.length,
     )[0];
 
-    const SecondaryNavigationClassName = classNames(
-      styles.SecondaryNavigation,
-      !icon && styles['SecondaryNavigation-noIcon'],
-    );
-
     secondaryNavigationMarkup = (
-      <div className={SecondaryNavigationClassName}>
-        <Secondary expanded={showExpanded} id={secondaryNavigationId}>
-          {subNavigationItems.map((item) => {
-            const {label, ...rest} = item;
-            const onClick = () => {
-              if (onNavigationDismiss) {
-                onNavigationDismiss();
-              }
-
-              if (item.onClick && item.onClick !== onNavigationDismiss) {
-                item.onClick();
-              }
-            };
-
-            return (
-              <Item
-                key={label}
-                {...rest}
-                label={label}
-                matches={item === longestMatch}
-                onClick={onClick}
-                truncateText={truncateText}
-              />
-            );
-          })}
-        </Secondary>
-      </div>
+      <SecondaryNavigation
+        ItemComponent={Item}
```

Comment:
Passing `Item` as prop to avoid circular dependencies. As long as the SecondaryNavigation component is not exposed to the consumers of the package and the API remains consistent, this approach should not cause any problems.